core/factorization/symbolic.cpp

```diff
@@ -58,45 +58,48 @@ GKO_REGISTER_OPERATION(cholesky_symbolic,
 GKO_REGISTER_OPERATION(prefix_sum, components::prefix_sum);
 GKO_REGISTER_OPERATION(initialize, lu_factorization::initialize);
 GKO_REGISTER_OPERATION(factorize, lu_factorization::factorize);
+GKO_REGISTER_HOST_OPERATION(compute_elim_forest, compute_elim_forest);
 
 
 }  // namespace
 
 
 /** Computes the symbolic Cholesky factorization of the given matrix. */
 template <typename ValueType, typename IndexType>
-std::unique_ptr<matrix::Csr<ValueType, IndexType>> symbolic_cholesky(
-    const matrix::Csr<ValueType, IndexType>* mtx)
+void symbolic_cholesky(
+    const matrix::Csr<ValueType, IndexType>* mtx,
+    std::unique_ptr<matrix::Csr<ValueType, IndexType>>& factors)
 {
     using matrix_type = matrix::Csr<ValueType, IndexType>;
     const auto exec = mtx->get_executor();
     const auto host_exec = exec->get_master();
-    const auto forest = compute_elim_forest(mtx);
+    std::unique_ptr<elimination_forest<IndexType>> forest;
+    exec->run(make_compute_elim_forest(mtx, forest));
```

Comment:
I don't know if we have thought of this yet, but this is a nested operation call. Can we handle that correctly in our loggers (or wherever that might be necessary)?

Reply:
The benchmark OperationLogger handles it correctly, with `-nested_names` resolving their nested structure. Similar things happen in kernels that call copy or create arrays. Profiling loggers will handle it correctly by design, since the annotation ranges are made for nesting.